Added 'do' and 'undo' commands.

lpatters opened the pull request:
Added commands to address issue 56.

lpatters committed:
Added 'do' and 'undo' commands.

jdc0589 commented:
@lpatters thanks for getting started on this.

Haven't not had a chance to review in detail yet, but it looks like we at least have a bunch of failing unit tests. The tests may likely need to be updated if there were breaking internal API changes.

jdc0589 commented:
*for reference: addresses #56*

lpatters committed:
Corrections.
Changes to permit 'mite up -f' to operate even when the 'dirty' state is detected.

lpatters committed:
Changes to provide more information for 'dirty' condition.
During an 'up --force' operation, 'dirty' actions which are ignored are reported.
